### PR TITLE
Allow the user to specify formats

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

Juju has a strong pattern of outputting different formats from each
command to allow the user to script against the CLI. The following
updates both find and info to allow this to happen.

The controversial thing about this change is naming the default output,
as for info the output is some sort of hybrid yaml that isn't yaml and
the default format for find is tabular. To be consistent I've named the
human (naming is hard).

The code is easy other than the warning logger to enable the commands to
output a warning if the result from the API isn't parsable. This can
happen if the charm creator fat fingers something and we don't want to
bring down the find/info commands to the user because of this. We'll
ignore the error (log it out) and present what we have to the user.

## QA steps

```sh
juju bootstrap lxd test
juju find wordpress
juju find --format=json wordpress
juju find --format=yaml wordpress
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/TBqoveMP/2786-charmhub-cli-formatting-options

File: cmd/juju/charmhub/find.go [hunk 1]
```diff
@@ -58,6 +61,11 @@ func (c *findCommand) Info() *cmd.Info {
 // It implements part of the cmd.Command interface.
 func (c *findCommand) SetFlags(f *gnuflag.FlagSet) {
 	c.ModelCommandBase.SetFlags(f)
+	c.out.AddFlags(f, "human", map[string]cmd.Formatter{
+		"yaml":  cmd.FormatYaml,
+		"json":  cmd.FormatJson,
+		"human": c.formatter,
```
Comment (hmlanigan) [inline cmd/juju/charmhub/find.go:67]: I like human, but suggest we add "tabular" here to keep in sync with other --format flags like in status

File: cmd/juju/charmhub/find_test.go [hunk 2]
```diff
@@ -35,11 +36,45 @@ func (s *findSuite) TestRun(c *gc.C) {
...
+	c.Assert(cmdtesting.Stdout(ctx), gc.Equals, `
+- type: object
+  id: charmCHARMcharmCHARMcharmCHARM01
+  name: wordpress
+  publisher: Wordress Charmers
+  summary: WordPress is a full featured web blogging tool, this charm deploys it.
+  version: 1.0.3
+  series:
+  - bionic
+  storeurl: https://someurl.com/wordpress
```
Comment (hmlanigan) [inline cmd/juju/charmhub/find_test.go:74]: We should be consistent.  The json and yaml output are different labels right now.  Most structs have it as "store-url"

File: cmd/juju/charmhub/info.go [hunk 3]
```diff
@@ -58,6 +62,11 @@ func (c *infoCommand) Info() *cmd.Info {
 // It implements part of the cmd.Command interface.
 func (c *infoCommand) SetFlags(f *gnuflag.FlagSet) {
 	c.ModelCommandBase.SetFlags(f)
+	c.out.AddFlags(f, "human", map[string]cmd.Formatter{
+		"yaml":  cmd.FormatYaml,
+		"json":  cmd.FormatJson,
+		"human": c.formatter,
```
Comment (hmlanigan) [inline cmd/juju/charmhub/info.go:68]: same as in find.

File: cmd/juju/charmhub/info_test.go [hunk 4]
```diff
@@ -34,11 +36,97 @@ func (s *infoSuite) TestRun(c *gc.C) {
...
+    provides:
+      source: dummy-token
+    requires:
+      sink: dummy-token
+  subordinate: false
+  usedby:
+  - wordpress-everlast
+  - wordpress-jorge
+  - wordpress-site
+bundle: null
```
Comment (hmlanigan) [inline cmd/juju/charmhub/info_test.go:116]: can we get an omitempty on bundle and charm?  maybe needs to be a pointer too?

File: cmd/juju/charmhub/info_test.go [hunk 5]
```diff
@@ -34,11 +36,97 @@ func (s *infoSuite) TestRun(c *gc.C) {
...
+type: charm
+id: charmCHARMcharmCHARMcharmCHARM01
+name: wordpress
+description: This will install and setup WordPress optimized to run in the cloud.
+publisher: Wordress Charmers
+summary: WordPress is a full featured web blogging tool, this charm deploys it.
+series:
+- bionic
+- xenial
+storeurl: https://someurl.com/wordpress
```
Comment (hmlanigan) [inline cmd/juju/charmhub/info_test.go:75]: same as with find

File: cmd/juju/charmhub/info_test.go [hunk 6]
```diff
@@ -34,11 +36,97 @@ func (s *infoSuite) TestRun(c *gc.C) {
...
+tags:
+- app
+- seven
+charm:
+  config:
+    options:
```
Comment (hmlanigan) [inline cmd/juju/charmhub/info_test.go:81]: maybe we don't need a --config TODO with this change.

Comment (SimonRichardson) [reply cmd/juju/charmhub/info_test.go:81]: Agreed

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**